src/pynwb/__init__.py

```diff
@@ -14,14 +13,65 @@
 from hdmf.build import BuildManager, TypeMap
 
 
+__rct_kwargs = list()
+
+
+# a function to register a container classes with the global map
+@docval({'name': 'neurodata_type', 'type': str, 'doc': 'the neurodata_type to get the spec for'},
+        {'name': 'namespace', 'type': str, 'doc': 'the name of the namespace'},
+        {"name": "container_cls", "type": type,
+         "doc": "the class to map to the specified neurodata_type", 'default': None},
+        is_method=False)
+def register_class(**kwargs):
+    """Register an NWBContainer class to use for reading and writing a neurodata_type from a specification
+    If container_cls is not specified, returns a decorator for registering an NWBContainer subclass
+    as the class for neurodata_type in namespace.
+    """
+    neurodata_type, namespace, container_cls = getargs('neurodata_type', 'namespace', 'container_cls', kwargs)
+
+    def _dec(cls):
+        __rct_kwargs.append({'data_type': neurodata_type, 'namespace': namespace, 'container_cls': cls})
+        return cls
```

Comment:
The above code will not work with extensions that call `register_class`. This will add kwargs to `__rct_kwargs` but they are not registered into the `__TYPE_MAP`. Same for the `register_map` method.

Reply:
Good catch... back to the drawing board.